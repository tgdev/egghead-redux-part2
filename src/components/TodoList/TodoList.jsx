import React from 'react';
import PropTypes from 'prop-types';

import Todo from 'components/Todo/Todo';

const TodoList = ({ todos, handleTodos }) => {
  return (
    <ul>
      {todos.map(todo => {
        return (
          <Todo
            key={todo.id}
            {...todo}
            handleTodo={() => handleTodos(todo.id)} />
        );
      })}
    </ul>
  );
};

TodoList.propTypes = {
  todos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      completed: PropTypes.bool.isRequired
    }).isRequired
  ).isRequired,
  handleTodos: PropTypes.func.isRequired
};

export default TodoList;
