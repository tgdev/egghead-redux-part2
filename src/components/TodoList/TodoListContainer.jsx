import { connect } from 'react-redux';

import { toggleTodo } from 'actions/todo.actions';

import getVisibleTodos from 'utils/get-visible-todos';

import TodoList from 'components/TodoList/TodoList';

const mapStateToProps = state => {
  return {
    todos: getVisibleTodos(state.todos, state.visibilityFilter)
  };
};

const mapDispatchToProps = dispatch => {
  return {
    handleTodos: id => {
      dispatch(toggleTodo(id));
    }
  };
};

const TodoListContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(TodoList);

export default TodoListContainer;
