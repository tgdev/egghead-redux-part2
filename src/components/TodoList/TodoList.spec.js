import React from 'react';
import { shallow } from 'enzyme';

import newTodosMock from 'mocks/new-todos.mock';

import TodoList from './TodoList';

describe('TodoList component', () => {
  const mockTodos = newTodosMock;
  const mockClickHandler = jest.fn();

  it('renders TodoList correctly', () => {
    const wrapper = shallow(<TodoList todos={mockTodos} handleTodos={mockClickHandler} />);
    expect(wrapper).toMatchSnapshot();
  });
});
