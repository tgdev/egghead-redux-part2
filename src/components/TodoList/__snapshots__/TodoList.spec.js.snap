// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component renders TodoList correctly 1`] = `
<ul>
  <Todo
    completed={false}
    handleTodo={[Function]}
    id="todo-id-0"
    key="todo-id-0"
    text="Learn Redux"
  />
  <Todo
    completed={false}
    handleTodo={[Function]}
    id="todo-id-1"
    key="todo-id-1"
    text="Master Redux"
  />
</ul>
`;
